content/300-guides/100-performance-and-optimization/150-connection-management/index.mdx

```diff
@@ -0,0 +1,286 @@
+
+#### Prevent hot reloading from creating new instances of `PrismaClient`
+
+Frameworks like [Next.js](https://nextjs.org/) support hot reloading of changed files, which enables you to see changes to your application without restarting. However, if the framework refreshes the module responsible for exporting `PrismaClient`, this can result in **additional, unwanted instances of `PrismaClient` in a development environment**.
+
+As a workaround, you can store `PrismaClient` as a global variable in development environments only, as global variables are not reloaded:
+
+```ts file=client.ts
+import { PrismaClient } from "@prisma/client";
+
+// add prisma to the NodeJS global type
+interface CustomNodeJsGlobal extends NodeJS.Global {
+    prisma: PrismaClient;
+}
+
+// prevent multiple instances of Prisma Client in development
+declare const global: CustomNodeJsGlobal;
+const prisma = global.prisma || new PrismaClient();
+if (process.env.NODE_ENV === "development") global.prisma = prisma;
+export default prisma;
+  prisma = global.prisma
+}
+
+export default prisma
+```
+
+The way that you import and use the client does not change:
+
+```ts file=app.ts
+import prisma from './client'
+
+async function main() {
+    const allUsers = await prisma.user.findMany();
+}
+
+main();
+```
+
+## Serverless environments (FaaS)
+
+Examples of serverless environments include Node.js functions hosted on AWS Lambda. Use the following checklist as a guide to connection management in serverless environments:
+
+- ☐ [Set connection limit](#recommended-connection-limit-1) based on whether you have an external connection pooler
+- ☐ Optionally [tune the pool size](#optimizing-the-connection-pool) if you are using an external connection pooler
+- ☐ [Instantiate `PrismaClient` outside the handler](#instantiate-prismaclient-outside-the-handler) and do not explicitly `$disconnect()`
+- ☐ Configure [function concurrency](#concurrency-limits) and handle [idle connections](#zombie-connections)
+
+### The serverless challenge
+
+In a serverless environment, each function creates **its own instance** of `PrismaClient`, and each client instance has its own connection pool. 
+
+Consider the following example, where a single AWS Lambda function uses `PrismClient` to connect to a database. The `connection_limit` is **3**:
+
+![An AWS Lambda function connecting to a database.](./serverless-connections.png)
+
+A traffic spike causes AWS Lambda to spawn two additional lambdas to handle the increased load. Each lambda creates an instance of `PrismaClient`, each with a `connection_limit` of **3**, which results in a maximum of **9** connections to the database:
+
+![Three AWS Lambda function connecting to a database.](./serverless-connections-2.png)
+ 
+200 _concurrent functions_ (and therefore 600 possible connections) responding to a traffic spike 📈 can exhaust the database connection limit very quickly.
```

Comment:
Maybe mention that functions that are paused, make this worse as they keep teh connection open by default and block them.